pkg/controller/job/job_controller.go

```diff
@@ -1693,11 +1696,32 @@ func findConditionByType(list []batch.JobCondition, cType batch.JobConditionType
 	return nil
 }
 
-func recordJobPodFinished(job *batch.Job, oldCounters batch.JobStatus) {
+func recordJobPodFinished(job, oldJob *batch.Job, newSucceededIndexes []int) {
 	completionMode := completionModeStr(job)
-	diff := job.Status.Succeeded - oldCounters.Succeeded
+	var diff int
+
+	// Updating succeeded metric must be handled differently
+	// for Indexed Jobs to handle the case where the job has
+	// been scaled down by reducing completions & parallelism
+	// in tandem, and now a previously completed index is
+	// now out of range (i.e. index >= spec.Completions).
+
+	if isIndexedJob(job) {
+		prevCompletedIndexes := succeededIndexesFromJob(oldJob)
+		newSucceeded := 0
+		for idx, _ := range uniqueIndexes(newSucceededIndexes) {
+			if !prevCompletedIndexes.has(idx) {
+				newSucceeded += 1
+			}
+		}
+		diff = newSucceeded
```

Comment:
Hmm orderedIntervals doesn't have a `len()` method but if I use Go's native 'len` function (i.e. len(succeededIndexesFromJob(...))` I believe it will return the number of intervals, not the number of completed indexes in those intervals - I don't think that is the desired behavior here, right? 

I think orderedIntervals's `.total()` method returns the number of completed indexes in the intervals, and I tried `succeededIndexes.total() - prevSucceededIndexes.total()` already which has issues described in another comment above.

Reply:
Ok I managed to avoid making the deep copy by doing some light refactoring to simply use the shallow copy of job.Status and the number of completions.

Reply:
oops, I meant total :)
I'll take another look